README.md

```diff
@@ -296,30 +296,34 @@ secrets:
     file: ./secrets/jcasc_vault
 ```
 #### Example: HashiCorp Vault Plugin as a Secret Source for JCasC
-Assume Vault has a secret at path `secret/jenkins/passwords` with keys `secretKey1` and `secretKey2`.  Set the value for environment variable `CASC_VAULT_PATHS` to `secret/jenkins/passwords`.
+Assume Vault has a secret at path `secret/jenkins/passwords` with keys `secretKey1` and `secretKey2`.  Set the value for environment variable `CASC_VAULT_PATHS` to `secret/jenkins`.
 The Hashicorp Vault Plugin provides two ways of accessing the secrets: using just the key within the secret and using the full path to the secret key.  The full path option allows for you to reference multiple secrets with overlapping keys.
 ```yaml
 credentials:
   system:
     domainCredentials:
       - credentials:
-          - string:
+          - vaultStringCredentialImpl:
               description: "Secret using only secret key name"
+              engineVersion: 2
               id: "secretUsingKey"
+              path: "${passwords}"
               scope: GLOBAL
-              token: "${secretKey1}"
-          - string:
+              vaultKey: "secretKey1"
+          - vaultStringCredentialImpl:
               description: "Secret using full path"
+              engineVersion: 2
               id: "secretUsingKey"
+              path: "${secret/jenkins/passwords}"
               scope: GLOBAL
-              token: "${secret/jenkins/passwords/secretKey1}"
+              vaultKey: "secretKey2"
 ```
```

Comment:
This goes against how JCasC vault hashicorp was intended to be used.

This doc explains how you can use JCasC secret source to embed secrets from vault.

Yes can also setup `vaultStringCredentialImpl` with JCasC secret source but that should be an addition to the readme.

Reply:
We have a test case: https://github.com/jenkinsci/hashicorp-vault-plugin/blob/master/src/test/java/com/datapipe/jenkins/vault/jcasc/secrets/VaultSecretSourceTest.java
That demonstrate the code base does indeed support this use case.